db/changes.go

```diff
@@ -1305,7 +1306,12 @@ func generateBlipSyncChanges(database *Database, inChannels base.Set, options Ch
 
 	// Store one-shot here to protect
 	isOneShot := !options.Continuous
-	err, forceClose = GenerateChanges(context.Background(), database, inChannels, options, docIDFilter, send)
+	// Use the context if provided in the options
+	ctx := options.ChangesCtx
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	err, forceClose = GenerateChanges(ctx, database, inChannels, options, docIDFilter, send)
```

Comment:
I don't think closing the context being passed to GenerateChanges is sufficient to terminate all goroutines associated with a changes feed.  Closing the changes terminator channel is the only thing that does that today - you can search for usages of that terminator to see where it's used - only one of those also does a check for the context being cancelled.  We could refactor changes to use a context throughout instead of the terminator channel, but otherwise I think we need to continue using the terminator to stop changes.